### ISSUE TITLE
Dev server doesn't watch changes in other packages

### ISSUE THREAD

**Issue by lionelhorn** [wxt-dev/wxt#428]
### Describe the bug
Dev server doesn't watch changes in code from other packages (but still internal to the repo)

### Expected behavior
Changes to be picked up as if they were in the same package.
React component changes from the same package are detected. 
Changes outside the extension package are not.

UI added as a content script

### Screenshots
https://github.com/wxt-dev/wxt/assets/50528276/8fecbe29-b462-4cd8-94c6-d01eefa8675e

**Comment by aklinker1** [comment]
I don't have enough info here, how are the packages connected, is this an integrated monorepo or a package based on? Can you share a reproduction? https://github.com/lionelhorn/wxt-wasm-vlcn-repro doesn't have the changes you show in the video

**Comment by lionelhorn** [comment]
Sorry, was cleaning the repro and got delayed.

```shell
git clone https://github.com/lionelhorn/wxt-wasm-vlcn-repro -b cs-no-watch
pnpm i
cd packages/ext
pnpm i
pnpm dev
```

Changing `packages/ext/src/components/HelloInADiv.tsx` will result in a reload.

Changing `packages/ui/src/lib/components/HelloInADivFromAnotherPackage.tsx` will not.

Import resolution via tsconfig alias 
https://github.com/lionelhorn/wxt-wasm-vlcn-repro/blob/e8689bce79cbf5b5ebda350b5799b1f1d38de18b/tsconfig.base.json#L19-L22

And vite plugin here:
https://github.com/lionelhorn/wxt-wasm-vlcn-repro/blob/e8689bce79cbf5b5ebda350b5799b1f1d38de18b/packages/ext/wxt.config.ts#L34-L35

**Comment by aklinker1** [comment]
I have a similar monorepo setup that does work. I'll check this out once I get back from traveling

**Comment by aklinker1** [comment]
Very interesting. I've looked into this. Vite, by default, watches files in 2 locations:

1. Any file loaded in the dev server
2. Any file inside the project root

In this example, since the shared component is only used in a content script, it's not loaded through the dev server. And since it's not inside the project root (`packages/ext`), it doesn't reload.

I'll add these types of files to the watcher 👍 

---

> For my future reference, here's the reproduction cleaned up to not have any type errors:
>
> [type-errors-gone.zip](https://github.com/wxt-dev/wxt/files/14288897/type-errors-gone.zip)
> 
> ```sh
> pnpm i
> pnpm --filter @test/ext dev
> ```

**Comment by aklinker1** [comment]
Released in [`v0.16.8`](https://github.com/wxt-dev/wxt/releases/tag/v0.16.8)

**Comment by aklinker1** [comment]
If you have a reproduction, feel free to open a new issue! It shouldn't have anything to do with the background, so it's probably a different issue.
